core/app/models/spree/payment_create.rb

```diff
@@ -3,16 +3,24 @@
 module Spree
   # Service object for creating new payments on an Order
   class PaymentCreate
+    PAYMENT_NOT_PROVIDED = Object.new.freeze
+
     # @param order [Order] The order for the new payment
     # @param attributes [Hash,ActionController::Parameters] attributes which are assigned to the new payment
     #   * :payment_method_id Id of payment method used for this payment
     #   * :source_attributes Attributes used to build the source of this payment. Usually a {CreditCard}
     #     * :wallet_payment_source_id (Integer): The id of a {WalletPaymentSource} to use
     # @param request_env [Hash] rack env of user creating the payment
     # @param payment [Payment] Internal use only. Instead of making a new payment, change the attributes for an existing one.
-    def initialize(order, attributes, payment: nil, request_env: {})
+    def initialize(order, attributes, payment: PAYMENT_NOT_PROVIDED, request_env: {})
       @order = order
-      @payment = payment
+
+      if payment != PAYMENT_NOT_PROVIDED
+        Spree::Deprecation.warn("Passing `payment:` to PaymentCreate.new is no longer supported.")
```

Comment:
Does this class continue to work exactly as before in this path? If yes I think this deprecation warning is misleading, seems like immediate action is needed. If we deprecate passing that parameter, we should say that its support will be removed in the future.

Reply:
Good call, I thought the deprecation horizon was being added by active support, changing that to use `deprecation_warning` which builds a better message:

```
DEPRECATION WARNING: Passing `payment:` to PaymentCreate.new is deprecated and will be removed from Solidus 4.0 (called from <main> at bin/rails:4)
```